File: lib/minicron/hub/assets/app/models/execution.js
'use strict';

(function() {
  Minicron.Execution = DS.Model.extend({
    number: DS.attr('number'),
    created_at: DS.attr('date'),
    started_at: DS.attr('date'),
    finished_at: DS.attr('date'),
    exit_status: DS.attr('number'),

    job: DS.belongsTo('job'),
    job_execution_outputs: DS.hasMany('job_execution_output', { async: true })
  });
})();

File: server/lib/minicron/hub/models/execution.rb
require Minicron::REQUIRE_PATH + 'hub/models/base'

module Minicron::Hub
  module Model
    class Execution < Minicron::Hub::Model::Base
      belongs_to :job, counter_cache: true
      belongs_to :host, counter_cache: true
      has_many :job_execution_outputs, dependent: :destroy
      has_many :alerts, dependent: :destroy

      validates :job_id, presence: true, numericality: { only_integer: true }
      validates :number, presence: true, numericality: { only_integer: true }
    end
  end
end
